src/west/app/main.py

```diff
@@ -644,10 +645,34 @@ def add_argument(self, *args, **kwargs):
         # Let argparse handle the actual argument.
         super().add_argument(*args, **kwargs)
 
+    def parse_known_args(self, args=None, namespace=None):
+        # argparse will terminate the application without raising an
+        # exception if parsing fails, making it hard to determine
+        # the cause of an error.
+        # This wrapper function records the reason for the error
+        # so we can display an appropriate error message in the
+        # error() function.
+        self.west_command_error = 'parse_error'
+        ret = super().parse_known_args(args, namespace)
+        self.west_command_error = None
+        return ret
+
     def error(self, message):
         if self.west_app and self.west_app.mle and \
            isinstance(self.west_app.mle, ManifestVersionError):
             log.die(mve_msg(self.west_app.mle))
+
+        if self.west_command_error == 'parse_error':
+            message += (
+                '\nPlease check the following:\n'
+                '- Check you entered the command properly\n'
+                '- If your command is not recognized, check you are in'
+                ' a valid `.west` workspace, and that the workspace '
```

Comment:
Maybe just call is `workspace`. After all the user has invoked the `west` command, and in that context the only `workspace` that exists is a `west` workspace.

```suggestion
                ' a workspace, and that the workspace '
```